weather_dl/download_pipeline/parsers.py

```diff
@@ -262,10 +330,10 @@ def process_config(file: t.IO) -> Config:
     """Read the config file and prompt the user if it is improperly structured."""
     config = parse_config(file)
 
-    def require(condition: bool, message: str) -> None:
-        """A assert-like helper that wraps text and throws a `ValueError`."""
+    def require(condition: bool, message: str, error_type: t.Any = ValueError) -> None:
```

Comment:
nit: can we have a more specific type for `error_type`? Will `Error` work here? Maybe you need `t.Type` [[1]].

[1]: https://docs.python.org/3/library/typing.html#typing.Type